### DISCUSSION TITLE
Model passes flopy check but crashes MF6 - looking for clues/suggestions

### DISCUSSION THREAD

**Discussion by wkitlasten** [MODFLOW-ORG/modflow6#2008] [General]
I'm using a scripted approach that has worked for dozens of models, but this one is causing issues that are not immediately obvious to me. It is a bit bigger than the others (2x820x531), but hopefully still within limits? I have no doubt the issues arise somewhere in the parameterizing with pyemu and some pest utilities magic (ensiprep, sva, etc). Just looking for clues/hints/suggestions of where to start looking since it worked with other models, the files look okay, and it passes the flopy check method.

Is there something in the arrays that would pass the flopy check but prevent MF6 from loading the model (NANs, domain issues, etc)

Cheers

> (py39) C:\worker_0>mf6
>                                    MODFLOW 6
>                 U.S. GEOLOGICAL SURVEY MODULAR HYDROLOGIC MODEL
>                         VERSION 6.4.1 Release 12/09/2022
> 
>    MODFLOW 6 compiled Dec 09 2022 18:18:37 with Intel(R) Fortran Intel(R) 64
>    Compiler Classic for applications running on Intel(R) 64, Version 2021.7.0
>                              Build 20220726_000000
> 
> This software has been approved for release by the U.S. Geological
> Survey (USGS). Although the software has been subjected to rigorous
> review, the USGS reserves the right to update the software as needed
> pursuant to further analysis and review. No warranty, expressed or
> implied, is made by the USGS or the U.S. Government as to the
> functionality of the software and related material nor shall the
> fact of release constitute any such warranty. Furthermore, the
> software is released on condition that neither the USGS nor the U.S.
> Government shall be held liable for any damages resulting from its
> authorized or unauthorized use. Also refer to the USGS Water
> Resources Software User Rights Notice for complete use, copyright,
> and distribution information.
> 
> 
>  Run start date and time (yyyy/mm/dd hh:mm:ss): 2024/08/29 19:35:42
> 
>  Writing simulation list file: mfsim.lst
>  Using Simulation name file: mfsim.nam
> 
> forrtl: error (65): floating invalid
> Image              PC                Routine            Line        Source
> mf6.exe            00007FF79B3D754C  IDMLOGGERMODULE_m         155  IdmLogger.f90
> mf6.exe            00007FF79B3C657F  LOADMF6FILETYPEMO         578  LoadMf6FileType.f90
> mf6.exe            00007FF79B3C3D57  LOADMF6FILETYPEMO         253  LoadMf6FileType.f90
> mf6.exe            00007FF79B3C3072  LOADMF6FILETYPEMO         151  LoadMf6FileType.f90
> mf6.exe            00007FF79B36E796  IDMMF6FILELOADERM          51  IdmMf6FileLoader.f90
> mf6.exe            00007FF79B1B792B  GWFNPFMODULE_mp_N         196  gwf3npf8.f90
> mf6.exe            00007FF79B1502E4  GWFMODULE_mp_GWF_         266  gwf3.f90
> mf6.exe            00007FF79B0D4E9D  SIMULATIONCREATEM         278  SimulationCreate.f90
> mf6.exe            00007FF79B0D26B1  SIMULATIONCREATEM         109  SimulationCreate.f90
> mf6.exe            00007FF79B0C1081  MF6COREMODULE_mp_          75  mf6core.f90
> mf6.exe            00007FF79B0C1042  MAIN__                     11  mf6.f90
> mf6.exe            00007FF79B4CB7BE  Unknown               Unknown  Unknown
> mf6.exe            00007FF79B4CBC69  Unknown               Unknown  Unknown
> mf6.exe            00007FF79B4CBB8E  Unknown               Unknown  Unknown
> mf6.exe            00007FF79B4CBA4E  Unknown               Unknown  Unknown
> mf6.exe            00007FF79B4CBCDE  Unknown               Unknown  Unknown
> KERNEL32.DLL       00007FF9BE097374  Unknown               Unknown  Unknown
> ntdll.dll          00007FF9BED9CC91  Unknown               Unknown  Unknown

It seems the issue is loading the npf package. The last bit of the .lst file is:

>  NPF -- NODE PROPERTY FLOW PACKAGE, VERSION 1, 3/30/2015 INPUT READ FROM UNIT 1008
> 
>  Loading input for NI/NPF
>  File generated by Flopy version 3.8.0.dev0 on 08/28/2024 at 16:56:40.
>    ICELLTYPE = 0

I checked all the arrays and input files manually, with numpy, and flopy and it all seems legit:

> >>> gwf.check()
> 
> ni MODEL DATA VALIDATION SUMMARY:
>   No errors or warnings encountered.
> 
>   Checks that passed:
>     npf package: zero or negative horizontal hydraulic conductivity values
>     npf package: horizontal hydraulic conductivity values below checker threshold of 1e-11
>     npf package: horizontal hydraulic conductivity values above checker threshold of 100000.0
>     coast package: BC indices valid
>     coast package: not a number (Nan) entries
>     coast package: BC in inactive cells
>     drn package: BC indices valid
>     drn package: not a number (Nan) entries
>     drn package: BC in inactive cells
> 
> flopy.utils.check.mf6check object at 0x000001E5B07FF550

**Comment by christianlangevin** [comment]
Hi @wkitlasten, looks like you are using an old version of MODFLOW 6.  I'd try the latest and see if that helps.  We've made quite a few changes and fixes.

**Reply by wkitlasten** [reply]
Doh, I had nans in my arrays!
